dlib/dnn/layers.h

```diff
@@ -21,6 +21,11 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
+    namespace impl
+    {
+        class visitor_fuse_convolutions;
+    }
+
```

Comment:
- Is it OK to put this here?
- Do you mind this class being a friend of the `con_` class?

Maybe I could move the definition of the `affine_` class here.

Reply:
I'm not super jazzed about friend functions like this.  It seems like it would be a lot simpler to add new layers entirely that support these fused operations and leave existing layers alone.